pkg/controllers/hpa/hpa_controller.go

```diff
@@ -46,7 +46,7 @@ func (c *HorizontalPodAutoscalerController) Reconcile(ctx context.Context, req c
 	if err := c.Client.Get(context.TODO(), req.NamespacedName, hpa); err != nil {
 		// The resource may no longer exist, in which case we delete related works.
 		if apierrors.IsNotFound(err) {
-			if err := c.deleteWorks(names.GenerateWorkName(util.HorizontalPodAutoscalerKind, req.Name, req.Namespace)); err != nil {
+			if err := c.deleteWorks(req.Namespace, names.GenerateWorkName(util.HorizontalPodAutoscalerKind, req.Name, req.Namespace)); err != nil {
```

Comment:
I'm not sure why this is needed, `req.Namespace` refers to hpa namespace but not the work namespace.
Actually the work name includes the workload(hpa) namespace info:
https://github.com/karmada-io/karmada/blob/1c1208d1394b8cee03a79acf6e22a25545e935d8/pkg/util/names/names.go#L62-L72